src/bp/ui-studio/src/web/views/OneFlow/index.tsx

```diff
@@ -162,6 +163,12 @@ const FlowBuilder = (props: Props) => {
     props.switchFlow(`${name}.flow.json`)
   }
 
+  const pathName = window.location.pathname.split('/')
+  const currentWorkflow = pathName.pop()
+  let currentTopic = pathName.pop()
+
+  currentTopic = currentTopic === 'oneflow' ? '' : currentTopic
+
```

Comment:
We should not rely on the URL to get that information, You already have access to the current flow name using the property `this.props.currentFlow`. In diagram/index.tsx: 

```js
const mapStateToProps = (state: RootReducer) => ({
  currentFlow: getCurrentFlow(state),
}


```

You could create a new selector in `selectors.ts` to return the current workflow name, and add one to extract the topic from the workflow (with a todo to replace with the helper function  https://github.com/botpress/botpress/pull/3272/files )

Reply:
It doesn't work when on a qna tho, I'm relying on the URL to show the qna form when the URL is `topic/qna` but the currentFlow is empty when that's the case